client/network/sync/src/state_request_handler.rs

```diff
@@ -144,7 +145,8 @@ where
 		);
 		protocol_config.inbound_queue = Some(tx);
 
-		let seen_requests = LruCache::new(num_peer_hint * 2);
+		let capacity = NonZeroUsize::new(num_peer_hint * 2).expect("cache capacity is not zero");
```

Comment:
the `num_peer_hint` is equivalent to `in_peers + out_peers`, so theoretically it can also be zero ?

Reply:
`num_peer_hint` is equivalent to `in_peers + out_peers`, so theoretically it can also be zero.

I am not sure which way to use, one is to specify `NonZeroUsize` type for `num_peer_hint` (but theoretically it can be zero when specifying the startup parameter), the other is to change `LruCache` to `Option<LruCache>`.